test/e2e/clusterctl_upgrade_test.go

```diff
@@ -144,7 +144,7 @@ var _ = Describe("When testing clusterctl upgrades (v1.6=>current)", func() {
 			InitWithBinary:            fmt.Sprintf(clusterctlDownloadURL, stableRelease),
 			InitWithProvidersContract: "v1beta1",
 			//  Note: Both InitWithKubernetesVersion and WorkloadKubernetesVersion should be the highest mgmt cluster version supported by the source Cluster API version.
-			InitWithKubernetesVersion: "v1.28.0",
+			InitWithKubernetesVersion: "v1.29.0",
```

Comment:
q: should we change also WorkloadKubernetesVersion? (same below)

Reply:
We have to wait until 1.6.1. before that it won't work as 1.6 doesn't have a mapping for the 1.29 image

Reply:
(xref last task in https://github.com/kubernetes-sigs/cluster-api/issues/9578#issuecomment-1853984699)